### PR TITLE
Adding Save Functionality for Match Detail Resolution Fields

### PR DESCRIPTION
**Pull request by ccaprio97**
## What’s changing?

_For External API changes, add the changelog tag to the PR and some CHANGELOG text here_

## Why?

_Link to specific issue (ex: Closes #123)_

## This PR has:

- [x] **Commented source code** (_required on public classes and methods, and elsewhere as appropriate_)

- [ ] **Developer documentation** (_for new build/test/API changes or complex portions of the system_)

- [x] **Automated unit tests** (_to maintain or increase level of code coverage_)

- [ ] **Changes to IAC** (_add any deployment steps that require manual intervention to the draft release notes_)

### CODE REVIEW

**Comment by ccaprio97** [comment]
POSSIBLE BUG - when testing on Friday after saving the resolution fields and then going back to do a query for a person the match id was new. 

This MAY be because the system was seeing all the resolution fields filled out and then closing the match record and so when searching for that match record again there were none that were OPEN so it created another. 

Must confirm before merging

**Comment by ccaprio97** [comment]
CSS still needs to be done too

**Comment by ccaprio97** [comment]
page also goes to match record search on save

**Comment by ccaprio97** [comment]
> page also goes to match record search on save

fixed

File: query-tool/tests/Piipan.QueryTool.Tests/MatchTest.cs [hunk 1]
```diff
@@ -244,7 +244,7 @@ public async Task TestSearchError_Post()
             pageModel.PageContext.HttpContext = contextMock();
 
             // act
-            Assert.IsType<PageResult>(await pageModel.OnPost());
+            Assert.IsType<PageResult>(await pageModel.OnPost(null));
```
Comment (dborgen2) [inline query-tool/tests/Piipan.QueryTool.Tests/MatchTest.cs:247]: Question: Do we need additional tests for when posting from this form? Looks like we're only testing the match search still.

File: shared/src/Piipan.Shared/Http/AuthorizedJsonApiClient.cs [hunk 2]
```diff
@@ -70,6 +70,26 @@ private async Task<HttpRequestMessage> PrepareRequest(string path, HttpMethod me
             return JsonConvert.DeserializeObject<TResponse>(responseContentJson);
         }
 
+        public async Task<TResponse> PatchAsync<TRequest, TResponse>(string path, TRequest body, Func<IEnumerable<(string, string)>> headerFactory)
```
[the remark this answers is not in the document]
Comment (ccaprio97) [reply shared/src/Piipan.Shared/Http/AuthorizedJsonApiClient.cs:73]: probably? I havent gotten the build to come back passed yet so I havent seen the test coverage yet

[the remark this answers is not in the document]
Comment (ccaprio97) [reply shared/src/Piipan.Shared/Http/AuthorizedJsonApiClient.cs:73]: ya need to add tests. coverage failing now

**Pull request merged**